### PR TITLE
Jee.20150713.ui fixes

### PR DESCRIPTION
**Pull request by jeeyoungk**
* added an error type to handle resource-limiting behavior (timeout, fetches, slots, etc)
* removed main/ui/ui.go-specific configs from ui/ui.go
* changed the error to use a monospace font.
* adding `application/json` header.

### CODE REVIEW

File: query/command.go [hunk 1]
```diff
@@ -143,7 +143,7 @@ func (cmd *SelectCommand) Execute(context ExecutionContext) (interface{}, error)
 			limitMessage = "the configured limit %d"
 		}
 		limitMessage = fmt.Sprintf(limitMessage, slotLimit)
-		return nil, fmt.Errorf("Requested number of data points (%d) exceeds %s", timerange.Slots(), limitMessage)
+		return nil, function.NewLimitError(fmt.Sprintf("Requested number of data points (%d) exceeds %s", timerange.Slots(), limitMessage), slotLimit)
```
Comment (achow) [inline query/command.go]: Interpolating `limitMessage` seems so awkward...

**Comment by achow** [comment]
To try to to standardize the limit messages, maybe we can set a `context` and a `limit` and build Error() with something like `"#{context}: limit exceeded (#{limit})"`? So a message might be `"fetch timeout: limit exceeded (60s)"`.

If we're creating special error types, it should have some kind of standard format.

File: main/common/common.go [hunk 2]
```diff
@@ -38,10 +38,16 @@ var (
 	Logger     = flag.String("logger", "glog", "Selects the logger to use")
 )
 
+type UIConfig struct {
+	Port      int `yaml:"port"`
```
Comment (achow) [inline main/common/common.go]: Why pull these fields out a level?

File: query/command.go [hunk 3]
```diff
@@ -143,7 +143,9 @@ func (cmd *SelectCommand) Execute(context ExecutionContext) (interface{}, error)
 			limitMessage = "the configured limit %d"
 		}
 		limitMessage = fmt.Sprintf(limitMessage, slotLimit)
```
Comment (achow) [inline query/command.go]: Should clean up `limitMessage` if we don't plan on using it anymore.

File: function/expression.go [hunk 4]
```diff
@@ -65,15 +65,26 @@ func (f MetricFunction) Evaluate(context EvaluationContext,
...
+		limit: n,
 	}
 }
 
+// limit returns the max # of fetches allowed by this counter.
+func (c FetchCounter) Limit() int {
+	return c.limit
+}
+
+func (c FetchCounter) Current() int {
```
Comment (achow) [inline function/expression.go]: What is this supposed to do? Looks like it's only going to be negative.

File: main/common/common.go (hunk 2 above)
Comment (jeeyoungk) [inline main/common/common.go]: there's two levels of configs.

1. running UI standalone (square/metrics) repo
2. running UI as part of an existing server (our internal repo).

these settings - port and timeout - are only required for (1) but the config was provided for both. i wanted to simplify the config from outside, thus split into two.

File: query/command.go (hunk 3 above)
Comment (jeeyoungk) [inline query/command.go]: removed these lines.

File: function/expression.go (hunk 4 above)
Comment (jeeyoungk) [inline function/expression.go]: i fixed it. it was supposed to return the current # of fetches. added tests for it.

**Comment by achow** [comment]
LGTM

**Pull request merged**